azerothcore/azerothcore-wotlk
# fix(Core/Players): Solve the issue with hunters never being able to r…

…oll need on polearms and properly check conditions to roll on relics



## Changes Proposed:
-  Check for item class when looking for shield conditions, as previously it conflicted with polearms (same subclass, different class)
-  Add conditions for totems/librams/idols as only their respective classes should be able to roll need on them
- This only applies to LFG loot rolls

## Issues Addressed:

- Closes https://github.com/chromiecraft/chromiecraft/issues/1794
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/8149

## SOURCE:


## Tests Performed:

- Tested needing polearms as hunters.
- Tested that non warriors/paladins/shamans still can't need on shields.
- Tested that non-shamans can't roll need on totems.
- The other items (sigils, librams, etc.) still need testing, as I don't really have it top of my mind any creatures in LFG dungeons that drop them to check, I'll  try to find them and update this later.
- 
## How to Test the Changes:


1. Create 5 characters, level them to 25 (one of them a hunter)
2. Queue Razorfen Kraul specific lfg dungeon
3. .go creature 4438 (Razorfen Spearhide, always drops a polearm)
4. Learn polearms on the hunter character
4. Check that you can need the item as a hunter


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
